lib/tapioca/sorbet_ext/name_patch.rb

```diff
@@ -6,7 +6,7 @@ module Types
     class Simple
       module NamePatch
         def name
-          @name ||= ::Tapioca::Reflection.name_of(@raw_type).freeze
+          @name ||= ::Tapioca::Reflection.qualified_name_of(@raw_type).freeze
```

Comment:
This just a stab at fixing which I think is directionally correct but
- doesn't totally solve the problems since I think not all types are `Simple`, some are just the classes.
- breaks a bunch of things 😬

Reply:
Thanks for the pointer, I'll take a look at that PR and play around with it a bit.

Reply:
@paracycle - I think I figured it out, see the latest commit ("Fix NamePatch"). By the time `NamePatch` gets applied, some types have already had their `name` memoized to the non-absolute path. Removing the memoization seems to fix the issue. LMK what you think.